### PR TITLE
refactor (jkube-kit/generator/webapp/): Migrate all tests from JMockit to Mockito

### PR DESCRIPTION
**Pull request by baruKreddy**
Fixes https://github.com/eclipse/jkube/issues/1596

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [ ] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] My code follows the style guidelines of this project
 - [ ] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/JettyAppSeverHandlerTest.java [hunk 1]
```diff
@@ -19,32 +19,35 @@
...
-
-import mockit.Expectations;
-import mockit.Mocked;
+import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.Assert.assertTrue;
```
Comment (rohanKanojia) [inline jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/JettyAppSeverHandlerTest.java:28]: Please use `assertThat(...).isTrue()` instead
```suggestion

```

File: jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/TomcatAppSeverHandlerTest.java [hunk 2]
```diff
@@ -19,32 +19,36 @@
...
 
 public class TomcatAppSeverHandlerTest {
 
   @Rule
   public TemporaryFolder temporaryFolder = new TemporaryFolder();
 
-  @Mocked
   private GeneratorContext generatorContext;
 
+  Plugin plugin;
```
Comment (rohanKanojia) [inline jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/TomcatAppSeverHandlerTest.java:40]:
```suggestion
  private Plugin plugin;
```

File: jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/TomcatAppSeverHandlerTest.java [hunk 3]
```diff
@@ -19,32 +19,36 @@
...
-
-import mockit.Expectations;
-import mockit.Mocked;
+import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.Assert.assertTrue;
```
Comment (rohanKanojia) [inline jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/TomcatAppSeverHandlerTest.java:28]: Please use `assertThat(...).isTrue()` instead
```suggestion

```

File: jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/WildFlyAppSeverHandlerTest.java [hunk 4]
```diff
@@ -22,33 +22,36 @@
...
-import mockit.Expectations;
-import mockit.Mocked;
+import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
```
Comment (rohanKanojia) [inline jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/WildFlyAppSeverHandlerTest.java:32]: Use `assertThat(..).isTrue()` `assertThat(..).isFalse()` `assertThat(..).isEqualTo()` instead
```suggestion
import static org.assertj.core.api.Assertions.assertThat;
```

File: jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/WildFlyAppSeverHandlerTest.java [hunk 5]
```diff
@@ -79,60 +78,44 @@ public void isApplicableHasJmsXmlAndThorntailPluginShouldReturnFalse(@Mocked Plu
...
-    // @formatter:on
+    when(generatorContext.getRuntimeMode()).thenReturn(RuntimeMode.KUBERNETES);
     // When
     final WildFlyAppSeverHandler handler = new WildFlyAppSeverHandler(generatorContext);
     // Then
     assertCommonValues(handler);
     assertEquals("jboss/wildfly:25.0.0.Final", handler.getFrom());
     assertEquals("/opt/jboss/wildfly/standalone/deployments", handler.getDeploymentDir());
-    assertTrue(handler.runCmds().isEmpty());
+    assertThat(handler.runCmds()).isEmpty();
```
Comment (rohanKanojia) [inline jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/WildFlyAppSeverHandlerTest.java:111]: Why haven't you replaced `assertEquals` with assertj?

File: jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/WildFlyAppSeverHandlerTest.java [hunk 6]
```diff
@@ -145,23 +128,20 @@ public void openShiftDockerStrategy() {
...
-    }};
-    // @formatter:on
+    when(generatorContext.getRuntimeMode()).thenReturn(RuntimeMode.OPENSHIFT);
+    when(generatorContext.getStrategy()).thenReturn(JKubeBuildStrategy.s2i);
     // When
     final WildFlyAppSeverHandler handler = new WildFlyAppSeverHandler(generatorContext);
     // Then
     assertCommonValues(handler);
     assertEquals("quay.io/wildfly/wildfly-centos7:26.0", handler.getFrom());
     assertEquals("/deployments", handler.getDeploymentDir());
```
Comment (rohanKanojia) [inline jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/WildFlyAppSeverHandlerTest.java:138]: Why haven't you replaced `assertEquals` with assertj?

File: jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/WildFlyAppSeverHandlerTest.java [hunk 7]
```diff
@@ -22,54 +22,53 @@
...
-import mockit.Expectations;
-import mockit.Mocked;
+import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
```
Comment (rohanKanojia) [inline jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/WildFlyAppSeverHandlerTest.java:32]: Delete these and use assertj 
```suggestion
```

File: jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/WildFlyAppSeverHandlerTest.java [hunk 8]
```diff
@@ -17,155 +17,131 @@
...
-    assertEquals("jboss:jboss:jboss", handler.getUser());
-    assertEquals(Collections.singletonMap("GALLEON_PROVISION_LAYERS", "cloud-server,web-clustering"), handler.getEnv());
-    assertEquals(Collections.singletonList("8080"), handler.exposedPorts());
+
+    assertThat(handler.supportsS2iBuild()).isTrue();
+    assertThat(handler.getAssemblyName()).isEqualTo("deployments");
+    assertThat(handler.getCommand()).isEqualTo("/opt/jboss/wildfly/bin/standalone.sh -b 0.0.0.0");
+    assertThat(handler.getUser()).isEqualTo("jboss:jboss:jboss");
+    assertThat(handler.getEnv()).isEqualTo(Collections.singletonMap("GALLEON_PROVISION_LAYERS", "cloud-server,web-clustering"));
+    assertThat(handler.exposedPorts()).isEqualTo(Collections.singletonList("8080")).isEqualTo( handler.exposedPorts());
```
Comment (rohanKanojia) [inline jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/handler/WildFlyAppSeverHandlerTest.java:145]: This doesn't look right

```suggestion
    assertThat(handler.exposedPorts()).isEqualTo(Collections.singletonList("8080"));
```

File: jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java [hunk 9]
```diff
@@ -92,7 +89,7 @@ public void customizeDoesNotSupportS2iBuildShouldThrowException() {
     // When
     new WebAppGenerator(generatorContext).customize(Collections.emptyList(), false);
     // Then - Exception thrown
-    fail();
+    fail("exception");
```
Comment (rohanKanojia) [inline jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java:92]: Could you please use `assertThatIllegalArgumentException` here?

```
    // When + Then
    assertThatIllegalArgumentException()
        .isThrownBy(() -> new WebAppGenerator(generatorContext).customize(Collections.emptyList(), false));
```

File: jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java [hunk 10]
```diff
@@ -240,9 +237,9 @@ public void extractEnvVariables_withMultipleEnvVariableWithSpacesSemicolonAndCom
     // when
     Map<String, String> extractedVariables = WebAppGenerator.extractEnvVariables(envConfig);
     // then
-    assertEquals("This is an environment variable with spaces", extractedVariables.get("ENV_WITH_SPACES"));
-    assertEquals("/path;/other/path", extractedVariables.get("ENV_WITH_SEMICOLON"));
-    assertEquals("layer1,layer2", extractedVariables.get("ENV_WITH_COMMAS"));
+    assertThat( extractedVariables.get("ENV_WITH_SPACES")).isEqualTo("This is an environment variable with spaces");
+    assertThat( extractedVariables.get("ENV_WITH_SEMICOLON")).isEqualTo("/path;/other/path");
+    assertThat(extractedVariables.get("ENV_WITH_COMMAS")).isEqualTo("layer1,layer2");
```
Comment (rohanKanojia) [inline jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java:242]:
```suggestion
    assertThat(extractedVariables)
        .containsEntry("ENV_WITH_SPACES", "This is an environment variable with spaces")
        .containsEntry("ENV_WITH_SEMICOLON", "/path;/other/path")
        .containsEntry("ENV_WITH_COMMAS", "layer1,layer2");
```

File: jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java [hunk 11]
```diff
@@ -228,7 +225,7 @@ public void extractEnvVariables_withSingleEnvVariable_shouldExtractIt() {
     // when
     Map<String, String> extractedVariables = WebAppGenerator.extractEnvVariables(envConfig);
     // then
-    assertEquals("web-server,ejb-lite,jsf,jpa,h2-driver", extractedVariables.get("GALLEON_PROVISION_LAYERS"));
+    assertThat(extractedVariables.get("GALLEON_PROVISION_LAYERS")).isEqualTo("web-server,ejb-lite,jsf,jpa,h2-driver");
```
Comment (rohanKanojia) [inline jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java:228]:
```suggestion
    assertThat(extractedVariables).containsEntry("GALLEON_PROVISION_LAYERS", "web-server,ejb-lite,jsf,jpa,h2-driver");
```

File: jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java [hunk 12]
```diff
@@ -217,8 +214,8 @@ public void extractEnvVariables_withMultipleEnvVariable_shouldExtractThemAll() {
     // when
     Map<String, String> extractedVariables = WebAppGenerator.extractEnvVariables(envConfig);
     // then
-    assertEquals("web-server,ejb-lite,jsf,jpa,h2-driver", extractedVariables.get("GALLEON_PROVISION_LAYERS"));
-    assertEquals("myproject", extractedVariables.get("IMAGE_STREAM_NAMESPACE"));
+    assertThat( extractedVariables.get("GALLEON_PROVISION_LAYERS")).isEqualTo("web-server,ejb-lite,jsf,jpa,h2-driver");
+    assertThat( extractedVariables.get("IMAGE_STREAM_NAMESPACE")).isEqualTo("myproject");
```
Comment (rohanKanojia) [inline jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java:218]: Use containsEntry here instead

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]

**Pull request merged**